buildSrc/src/main/java/com/example/yandex/practicum/middle_homework_5/FindUntranslatedStringsPlugin.kt

```diff
@@ -0,0 +1,99 @@
+package com.example.yandex.practicum.middle_homework_5
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.GradleException
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.tasks.TaskAction
+import java.io.File
+import java.io.FilenameFilter
+import javax.xml.parsers.DocumentBuilderFactory
+
+
+class FindUntranslatedStringsPlugin : Plugin<Project> {
+    override fun apply(project: Project) {
+        project.tasks.register(TASK_NAME, FindUntranslatedStringsTask::class.java)
+    }
+
+    companion object {
+        private const val TASK_NAME = "untranslatedStrings"
+    }
+}
+
+abstract class FindUntranslatedStringsTask : DefaultTask() {
+    @TaskAction
+    fun findUntranslatedStrings() {
+        val resDir = File(project.projectDir, DIR_PATH)
+        val baseStrings = File(resDir, RES_DIR)
+
+        val stringValues = parseStrings(baseStrings)
+
+        val dirFilter = FilenameFilter { dir, name ->
+            val file = File(dir, name)
+            file.isDirectory && file.name.startsWith("values-")
+        }
+
+        val missingTranslations = mutableMapOf<String, List<String>>()
+
+        val localizedDirs = resDir.listFiles(dirFilter)
+        if (localizedDirs.isNullOrEmpty() || stringValues.isEmpty()) {
+            throw GradleException(String.format(NO_RES_DIRS_MSG, project.name))
+        }
+
+        localizedDirs?.forEach { directory ->
+            val translationsFile = File(directory, RES_FILE_NAME)
+            if (translationsFile.exists()) {
+                val translationStrings = parseStrings(translationsFile)
+                val untranslatedStrings = mutableListOf<String>()
+                untranslatedStrings.addAll(stringValues.subtract(translationStrings))
+                if (untranslatedStrings.isNotEmpty()) {
+                    missingTranslations[directory.absolutePath] = untranslatedStrings
+                }
```

Comment:
Здесь можно проще сделать:
```suggestion
                val untranslatedStrings = stringValues.subtract(translationStrings)
                if (untranslatedStrings.isNotEmpty()) {
                    missingTranslations[directory.absolutePath] = untranslatedStrings
                }
```